pkg/prometheus/operator.go

```diff
@@ -1629,6 +1635,7 @@ func (c *Operator) createOrUpdateConfigurationSecret(ctx context.Context, p *mon
 		additionalAlertRelabelConfigs,
 		additionalAlertManagerConfigs,
 		ruleConfigMapNames,
+		c.endpointSliceSupported,
```

Comment:
The ConfigGenerator struct could have a new `enableEndpointSlices` bool field so we don't have to pass the flag around. Another advantage is that we can also check that the Prometheus version supports the endpointslice role.